### DISCUSSION TITLE
BinSPreader to refine MAGs from vamb using individually assembled samples

### DISCUSSION THREAD

**Discussion by eperezv** [ablab/spades#1023] [Q&A] [answered]
Hello,

I came across the BinSPreader tool and manuscript today while looking for ways to refine MAGs, and in particular thinking about how this may affect my dataset and the detection of MGEs in MAGs.

I am interested in giving a try to this tool but the approach I followed before (vamp on samples assembled individually as vamb coders recommends) seems incompatible with BinSPreader. Basically, what I have tried before is to assemble samples individually, merge the contigs and get MAGs with vamb, metabat2 and maxbin2 that I analyze with DAS tool. However, the approach recommended by BinSPreader seems to need a single gfa file (I guess from a co-assembly).
  
Is it possible (or even recommended) to run BinSPreader on different assemblies and MAGs obtained from them? What would be your recommendation in this case?

Thank you,
Eduardo

**Comment by asl** [comment, marked as answer]
Well, BinSPreader relies on assembly graph and in your case there is no single assembly graph to use for the refinement. I must also note that the co-binning procedure might lead to the high duplication ratios in MAGs especially if multiple strains are present. So, likely your only way is:
  - Co-assemble everything
  - Take the scaffolds from this co-assembly
  - Bin them and using BinSPreader for the refinement

Alternatively:
  - Assemble your samples individually
  - Bin individual samples separately
  - Use BinSPreader to refine them
  - Use DAS_Tool / dRep to dereplicate MAGs. Be careful as the tools might drop short contigs that come from the elements you've interested in

**Reply by eperezv** [reply]
Thanks a lot for the help. Co-assembly seems to not work well for my dataset, as the number of bins that I recover with different methods is quite low as compared to individual samples. I will give a try to the the alternative.

**Comment by SilasK** [comment]
As I understand, you should be able to refine bins with binspreader for each sample individually. It doesn't matter which bin-workflow generated the contig-bin table, as long as you only have contigs from the sample.

Or am I wrong?

**Reply by asl** [reply]
That's correct. You need to know contig paths through the assembly graph though. So, if the graph was generated by SPAdes you'd need to bin scaffolds or, alternatively, explicitly provide contig paths
